### PR TITLE
feature(build): update gradle from 4.0 to 6.3

### PR DESCRIPTION
**Pull request by 317787106**
**What does this PR do?**

- update gradle from 4.0 to 6.3
- update lombok, etc.
- delete some unuseful dependency.

**Why are these changes required?**

**This PR has been tested by:**
- Unit Tests
- Manual Testing

**Follow up**

**Extra details**

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -43,59 +46,57 @@ buildscript {
...
-        protobufVersion = '3.21.12'
-        protobufGradlePluginVersion = '0.8.0'
-    }
 
     dependencies {
-        classpath 'com.github.jengelman.gradle.plugins:shadow:2.0.2'
-        classpath 'com.google.protobuf:protobuf-gradle-plugin:0.8.3'
-        classpath "gradle.plugin.com.liferay:gradle-plugins-node:4.3.0"
+        classpath 'com.github.jengelman.gradle.plugins:shadow:5.2.0'
+        classpath 'com.google.protobuf:protobuf-gradle-plugin:0.8.12'
```
Comment (lxcmyf) [inline build.gradle:52]: In line 51, this version is 7.1.2 in java-tron.

File: build.gradle [hunk 2]
```diff
@@ -43,59 +46,57 @@ buildscript {
...
+    implementation group: 'com.alibaba', name: 'fastjson', version: '1.2.83'
+    implementation group: 'com.fasterxml.jackson.core', name: 'jackson-databind', version: '2.16.1'
+
+    compileOnly 'org.projectlombok:lombok:1.18.24'
+    annotationProcessor 'org.projectlombok:lombok:1.18.24'
+    testCompileOnly 'org.projectlombok:lombok:1.18.24'
+    testAnnotationProcessor 'org.projectlombok:lombok:1.18.24'
+
+    implementation group: 'org.jline', name: 'jline', version: '3.25.0'
+    implementation group: 'io.github.tronprotocol', name: 'zksnark-java-sdk', version: '1.0.0'
```
Comment (lxcmyf) [inline build.gradle:89]: It's best to keep these upgraded versions consistent with Java tron.

File: build.gradle (hunk 1 above)
Comment (317787106) [reply build.gradle:52]: which dependency ?

File: build.gradle (hunk 2 above)
Comment (317787106) [reply build.gradle:89]: The version of grpc、protobuf is same as that used in java-tron. Other dependency is not very important.

File: build.gradle (hunk 1 above)
Comment (317787106) [reply build.gradle:52]: classpath "gradle.plugin.com.github.johnrengelman:shadow:7.1.2" . This version of Shadow supports Gradle 7.0+ only. So 5.2.0 is available.

**Comment by Chator1** [comment]
Re: [actions/checkout] 0xFD689e5f2d8d9Aec0aD328225Ae62FdBDdb30328 (Issue #1918) (رد: [الإجراءات/الدفع] 0xFD689e5f2d8d9Aec0aD328225Ae62FdBDdb30328 (الإصدار رقم 1918))

File: gradle/wrapper/gradle-wrapper.properties [hunk 3]
```diff
@@ -3,4 +3,4 @@ distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
 zipStoreBase=GRADLE_USER_HOME
 zipStorePath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-4.0-all.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-6.3-all.zip
```
Comment (halibobo1205) [inline gradle/wrapper/gradle-wrapper.properties:6]: use `./gradlew wrapper --gradle-version=6.3` ?

File: build.gradle [hunk 4]
```diff
@@ -114,6 +115,10 @@ protobuf {
     }
 }
 
+clean.doFirst {
+    delete "src/main/gen"
+}
+
```
Comment (Chator1) [inline build.gradle:121]: 0xFD689e5f2d8d9Aec0aD328225Ae62FdBDdb30328

**Review by Chator1 — COMMENTED** [review]
Binance wallet (0xFD689e5f2d8d9Aec0aD328225Ae62FdBDdb30328) Email bdalhafzslah382@gmail.com

File: gradle/wrapper/gradle-wrapper.properties (hunk 3 above)
Comment (317787106) [reply gradle/wrapper/gradle-wrapper.properties:6]: what do you mean?

Comment (317787106) [reply gradle/wrapper/gradle-wrapper.properties]: I have updated the gradle related file.

File: build.gradle (hunk 1 above)
Comment (317787106) [reply build.gradle:52]: @lxcmyf shadow is not necessary for java-tron. You can try to compile it without shadow.

**Review by lxcmyf — APPROVED** [review]

**Review by halibobo1205 — APPROVED** [review]

**Review by hasti0073 — APPROVED** [review]

**Pull request merged**